src/java.base/share/classes/java/lang/foreign/ValueLayout.java

```diff
@@ -414,27 +417,29 @@ sealed interface OfAddress extends ValueLayout permits ValueLayouts.OfAddressImp
         OfAddress withOrder(ByteOrder order);
 
         /**
-         * Returns an <em>unbounded</em> address layout with the same carrier, alignment constraint, name and order as this address layout,
-         * but with the specified pointee layout. An unbounded address layout allow raw addresses to be accessed
-         * as {@linkplain MemorySegment memory segments} whose size is set to {@link Long#MAX_VALUE}. As such,
-         * these segments can be used in subsequent access operations.
+         * Returns an address layout with the same carrier, alignment constraint, name and order as this address layout,
+         * but associated with the specified target layout. The returned address layout allows raw addresses to be accessed
+         * as {@linkplain MemorySegment memory segments} whose size is set to the size of the specified layout. Moreover,
+         * if the accessed raw address is not compatible with the alignment constraint in the provided layout,
+         * {@linkplain IllegalArgumentException} will be thrown.
          * <p>
          * This method is <a href="package-summary.html#restricted"><em>restricted</em></a>.
          * Restricted methods are unsafe, and, if used incorrectly, their use might crash
          * the JVM or, worse, silently result in memory corruption. Thus, clients should refrain from depending on
          * restricted methods, and use safe and supported functionalities, where possible.
          *
-         * @return an unbounded address layout with same characteristics as this layout.
+         * @param layout the target layout.
+         * @return an address layout with same characteristics as this layout, but with the provided target layout.
          * @throws IllegalCallerException If the caller is in a module that does not have native access enabled.
-         * @see #isUnbounded()
+         * @see #targetLayout()
          */
         @CallerSensitive
-        OfAddress asUnbounded();
+        OfAddress withTargetLayout(MemoryLayout layout);
```

Comment:
I think it could still be nice to have `asUnbounded` as a shortcut for `withTargetLayout(sequenceLayout(JAVA_BYTE))`.

WDYT?

Reply:
Perhaps we can leave it - but I'm on the fence, as I think having too many options can also be confusing (esp. given that the options are restricted) - should then we also add slicing methods which just remove spatial bounds? Also, it's probably a bit of a naming issue as well - e.g. if the two methods were telescopic, with same name - but one had no argument and can be explained as calling the other, I'd be more ok with it. But having both `withTargetLayout` *and* `asUnbounded` feels like you can do two very different things, as the names are unrelated.

Reply:
I've decided to just focus on the primitive operations for now. New overloads can be added as required, even at a later point.